GitUI/UserControls/RevisionGrid.cs

```diff
@@ -1101,7 +1099,14 @@ public void ForceRefreshRevisions()
                 _filtredCurrentCheckout = null;
                 _currentCheckoutParents = null;
                 SuperprojectCurrentCheckout = newSuperPrjectInfo;
-                ChangedFiles = changedFilesTask;
+                if (ArtificialCountEnabled)
+                {
+                    ChangedFiles = Task.Run(() => capturedModule.GetAllChangedFiles());
```

Comment:
This isn't something directly introduced in this PR, but it touches the problematic code.
The problem is related to non-thread-safety of GitModule (related discussions https://github.com/gitextensions/gitextensions/pull/4164#issuecomment-345519886). 

The `capturedModule` on line 1079 is instantiated incorrectly, a correct implementation will look something like https://github.com/gitextensions/gitextensions/blob/c133bf68944be5e68fcdc66dc7dcdc4c792c222e/GitUI/CommandsDialogs/FormBrowse.cs#L2417-L2422 
(albeit in this case a copy would be init'ed outside the capture)

Reply:
Actually, ignore this.
Reviewing a different PR, I realised that this construct makes things non unit testable....
This and other instances will need to be dealt with separately...

Reply:
I plan to remove this task, get the count together with the commit button. The SubjectCount split is the first step. I introduced it here because it simplified the change slightly. 
Can the implementation be left as is?

Reply:
Yes, leave it as is.